Eye Module Changes Issue #1233 and Issue #1159

sjpadgett opened the pull request:
Convert to mPDF library.
Fix for pdf generation.
Style changes.

(below added by bradymiller)
demo is here: http://www.open-emr.org/wiki/index.php/Development_Demo#Epsilon_-_Up_For_Grabs_Demo

sjpadgett commented:
@ophthal Let me know if I've violated your hard work. Module looks great, nice job. Seems to be some missing icon's in image directory and styles and report.css have/had numerous syntax problems which I fixed most but the nested css classes need addressing. Supposedly css3 supports but i'm not so sure.
I also changed the fixed nav to work so now you will only get menu button when viewport is less than 600, I believe. Certainly won't hurt my feelings if i've done something you don't like, just let me know.
@bradymiller , can we now get CCM addressed so we can release v5.0.1 and move on :)  Would be great if you could get on my up-for or a demo so Ray can take a look.

MatthewVita commented:
LGTM

bradymiller commented:
Up for grabs demo for this is here: http://www.open-emr.org/wiki/index.php/Development_Demo#Epsilon_-_Up_For_Grabs_Demo

bradymiller commented:
When try to save PDF, nothing happens and get following php error:
``````
[19-Nov-2017 00:20:08 Europe/London] PHP Fatal error:  Uncaught MpdfException: Unable to create output file: /tmpReport_2.pdf in /var/www/html/b/openemr/vendor/mpdf/mpdf/mpdf.php:9443
Stack trace:
#0 /var/www/html/b/openemr/interface/forms/eye_mag/php/taskman_functions.php(498): mPDF->Output('/tmpReport_2.pd...', 'F')
#1 /var/www/html/b/openemr/interface/forms/eye_mag/php/taskman_functions.php(113): make_document(Array)
#2 /var/www/html/b/openemr/interface/forms/eye_mag/taskman.php(118): process_tasks(Array)
#3 {main}
  thrown in /var/www/html/b/openemr/vendor/mpdf/mpdf/mpdf.php on line 9443
``````

bradymiller commented:
Guessing above error will be an easy fix.
The optho form looks great though now in all themes. The eye module is back in business for 5.0.1!

bradymiller commented:
In the error reported above, note I just added another line to the top of it.

sjpadgett commented:
That's telling me I can't open the system temp folder i.e sys_get_temp_dir(). Here I thought I was being clever! Guess I'll use documents temp....

sjpadgett commented:
Besides the temp dir issue the File->Save Report as Pdf saves pdf to patient document directory however doesn't show up in demographics documents. Also, doesn't ask to save but simply tells user it is saved so I assume this is meant to be, guess I need to fix why not showing in patients documents.
Wonder where the other save to pdf get's invoked (in save.php)?

sjpadgett commented:
@bradymiller We can see if this directory takes on Docker.

bradymiller commented on interface/forms/eye_mag/php/taskman_functions.php:
```diff
@@ -466,8 +486,15 @@ function make_document($task)
+
+    $pdf->writeHTML($content, 0); // false or zero works for both mPDF and HTML2PDF
+
+    $tmpdir = $GLOBALS['OE_SITE_DIR'] . '/documents/temp/'; // Best to get a known system temp directory to ensure a writable directory.
```
Is this file getting removed after it's used?

bradymiller commented on interface/forms/eye_mag/save.php:
```diff
@@ -386,9 +406,14 @@
+        //$pdf->writeHTML($content, 2);
+
+        $pdf->writeHTML($content, false); // false or zero works for both mPDF and HTML2PDF
+        $tmpdir = $GLOBALS['OE_SITE_DIR'] . '/documents/temp/'; // Best to get a known system temp directory to ensure a writable directory.
```
Is this file getting removed after it's used?

sjpadgett replied:
Nope, thats why I sent to system. I'm not even sure it needs to get saved at all.

bradymiller replied:
Very odd. It should not linger (note the filename is not very specific, so could chance it would get overwritten).  It looks like it's supposed to store it has a document, but it doesn't seem to work on the testing I just did.

bradymiller replied:
@sjpadgett ,
It appears you may be getting sucked into another OpenEMR vortex. I wish you the best :)
-brady

sjpadgett replied:
I'll delete them. just tested..need a minute

sjpadgett replied:
Oh well, this is a very nice module so worth the effort.

sjpadgett commented:
Found why not showing in documents->encounters. $category_id is set but for some reason is out of scope by time gets to save. Trying to determine if should fetch again or persist when set.

sjpadgett commented:
@bradymiller This should get us to closing the pdf issue if the temp directory writes and cleanup are working. Once again, you are a patient man. Thanks.

bradymiller commented:
Nice work. This also fixes the font size issue. Will bring this into codebase. thanks! -brady

sjpadgett commented:
Just a final note. We should really advertise this feature more. I still haven't been through all it can do. Really great work Ray...

ophthal commented:
@sjpadgett Glad to have the help.  I’ll look at the demo.  There has always been a font issue with this form that’s why it only looked right in tan.  It works for me, and now it may work for others too…

I have added more functionality to this form that I should probably add to the codebase.  Little things that make my charting faster like a mini-Fee Sheet that fits in the Coding Engine area.  Some of it needs to be replaced, like the  Imaging Library. Maybe there is a way to replace it with the PACS that is being added… We’ll see.

The problem with testing this form is that a lot of the advanced functionality doesn’t become evident until you actually use it:  you need to have historical visit data for more than 10 encounters a month, referring providers in address book, ICD10 and CPT4 loaded, a fax server, etc…  Without these, some functions are not apparent and cannot be tested:  PMH “lists” are provider-specific based on time, Glaucoma Flow Sheet, Impression/Plan Builder, Coding Engine, Prior Visit data, Copy forward, Communication Engine.  The Shorthand entry technique only makes sense after seeing it in action - only an eye doc would think it was beautiful.  And wow is it fast.  Only way to make it faster is to add voice…  Hmm, I see another project in my future…

Anyway, thanks for your help.  It is truly appreciated.  

> On Nov 18, 2017, at 1:03 PM, Jerry Padgett <notifications@github.com> wrote:
> 
> @ophthal <https://github.com/ophthal> Let me know if I've violated your hard work. Module looks great, nice job. Seems to be some missing icon's in image directory and styles and report.css have/had numerous syntax problems which I fixed most but the nested css classes need addressing. Supposedly css3 supports but i'm not so sure.
> I also changed the fixed nav to work so now you will only get menu button when viewport is less than 600, I believe. Certainly won't hurt my feelings if i've done something you don't like, just let me know.
> @bradymiller <https://github.com/bradymiller> , can we now get CCM addressed so we can release v5.0.1 and move on :) Would be great if you could get on my up-for or a demo so Ray can take a look.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/openemr/openemr/pull/1238#issuecomment-345459484>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAfDcG82Ee69hqe7d-iA6FekXtCtlZqtks5s3xvpgaJpZM4QjFbI>.
>

sjpadgett commented:
Glad I could help. We will certainly keep an EYE :) out for new features..

On Sun, Nov 19, 2017 at 7:10 AM, Ray Magauran <notifications@github.com>
wrote:

> @sjpadgett Glad to have the help. I’ll look at the demo. There has always
> been a font issue with this form that’s why it only looked right in tan. It
> works for me, and now it may work for others too…
>
> I have added more functionality to this form that I should probably add to
> the codebase. Little things that make my charting faster like a mini-Fee
> Sheet that fits in the Coding Engine area. Some of it needs to be replaced,
> like the Imaging Library. Maybe there is a way to replace it with the PACS
> that is being added… We’ll see.
>
> The problem with testing this form is that a lot of the advanced
> functionality doesn’t become evident until you actually use it: you need to
> have historical visit data for more than 10 encounters a month, referring
> providers in address book, ICD10 and CPT4 loaded, a fax server, etc…
> Without these, some functions are not apparent and cannot be tested: PMH
> “lists” are provider-specific based on time, Glaucoma Flow Sheet,
> Impression/Plan Builder, Coding Engine, Prior Visit data, Copy forward,
> Communication Engine. The Shorthand entry technique only makes sense after
> seeing it in action - only an eye doc would think it was beautiful. And wow
> is it fast. Only way to make it faster is to add voice… Hmm, I see another
> project in my future…
>
> Anyway, thanks for your help. It is truly appreciated.
>
> > On Nov 18, 2017, at 1:03 PM, Jerry Padgett <notifications@github.com>
> wrote:
> >
> > @ophthal <https://github.com/ophthal> Let me know if I've violated your
> hard work. Module looks great, nice job. Seems to be some missing icon's in
> image directory and styles and report.css have/had numerous syntax problems
> which I fixed most but the nested css classes need addressing. Supposedly
> css3 supports but i'm not so sure.
> > I also changed the fixed nav to work so now you will only get menu
> button when viewport is less than 600, I believe. Certainly won't hurt my
> feelings if i've done something you don't like, just let me know.
> > @bradymiller <https://github.com/bradymiller> , can we now get CCM
> addressed so we can release v5.0.1 and move on :) Would be great if you
> could get on my up-for or a demo so Ray can take a look.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub <
> https://github.com/openemr/openemr/pull/1238#issuecomment-345459484>, or
> mute the thread <https://github.com/notifications/unsubscribe-
> auth/AAfDcG82Ee69hqe7d-iA6FekXtCtlZqtks5s3xvpgaJpZM4QjFbI>.
>
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openemr/openemr/pull/1238#issuecomment-345511788>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABxhkSIZGedNhOsc5pjI_Ma2Fd0ZM7Llks5s4BqygaJpZM4QjFbI>
> .
>



-- 
Jerry Padgett
Padgett's Consulting
Brandon, Florida 33511
sjpadgett@gmail.com <sjpadgett@gmailo.com>
